storage/innobase/row/row0sel.cc

```diff
@@ -4009,6 +4009,7 @@ dberr_t row_search_no_mvcc(byte *buf, page_cur_mode_t mode,
   cursor is open to return all the records. */
   if (direction != 0) {
     if (prebuilt->m_temp_read_shared && !prebuilt->m_temp_tree_modified) {
+      // QQ: Should we check mtr_records on this branch as well?
```

Comment:
tree is not modified, will we use this cached cursor and keep mtr active here?

I think its not required but please check a bit

Reply:
As discussed on Slack it turns out we need to periodically commit mtr in this case as well.
Added test case for this scenario.